### PR TITLE
smp server: log prometheus metrics

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    info =
+      "# Info\n\
+      \# ----\n\
+      \\n"
+      <> socketsMetric socketsAccepted "simplex_smp_sockets_accepted" "Accepted sockets"
+      <> socketsMetric socketsClosed "simplex_smp_sockets_closed" "Closed sockets"
+      <> socketsMetric socketsActive "simplex_smp_sockets_active" "Active sockets"
+      <> socketsMetric socketsLeaked "simplex_smp_sockets_leaked" "Leaked sockets"
+      <> "# HELP simplex_smp_threads_total Threads\n\
+      \# TYPE simplex_smp_threads_total gauge \n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:347]:
```suggestion
      \# TYPE simplex_smp_threads_total gauge\n\
```

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+      \# TYPE simplex_smp_subscription_ntf_total gauge\n\
+      \simplex_smp_subscription_ntf_total " <> tshow ntfSubsCount <> "\n# ntfSubs\n\
+      \\n\
+      \# HELP simplex_smp_subscription_ntf_clients_total Total subscribed NTF servers, first counting method\n\
+      \# TYPE simplex_smp_subscription_ntf_clients_total gauge\n\
+      \simplex_smp_subscription_ntf_clients_total " <> tshow ntfSubClientsCount <> "\n# ntfSubClients\n"
+    socketsMetric :: (SocketStats -> Int) -> Text -> Text -> Text
+    socketsMetric sel metric descr =
+      "# HELP " <> metric <> " " <> descr <> "\n"
+        <> "# TYPE " <> metric <> " gauge \n"
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:372]:
```suggestion
        <> "# TYPE " <> metric <> " gauge\n"
```

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+      <> socketsMetric socketsAccepted "simplex_smp_sockets_accepted" "Accepted sockets"
+      <> socketsMetric socketsClosed "simplex_smp_sockets_closed" "Closed sockets"
+      <> socketsMetric socketsActive "simplex_smp_sockets_active" "Active sockets"
+      <> socketsMetric socketsLeaked "simplex_smp_sockets_leaked" "Leaked sockets"
+      <> "# HELP simplex_smp_threads_total Threads\n\
+      \# TYPE simplex_smp_threads_total gauge \n\
+      \simplex_smp_threads_total " <> tshow threadsCount <> "\n\
+      \\n\
+      \# HELP simplex_smp_clients_total Clients\n\
+      \# TYPE simplex_smp_clients_total gauge \n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:351]:
```suggestion
      \# TYPE simplex_smp_clients_total gauge\n\
```

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+      \simplex_smp_queues_notify_deleted " <> tshow _ntfDeleted <> "\n# ntfDeleted\n\
+      \\n\
+      \# HELP simplex_smp_queues_notify_deleted_batch Deleted batched queues with notification flag (client).\n\
+      \# TYPE simplex_smp_queues_notify_deleted_batch counter\n\
+      \simplex_smp_queues_notify_deleted_batch " <> tshow _ntfDeletedB <> "\n# ntfDeletedB\n\
+      \\n\
+      \# HELP simplex_smp_queues_notify_total Total number of stored queues with notification flag.\n\
+      \# TYPE simplex_smp_queues_notify_total gauge\n\
+      \simplex_smp_queues_notify_total1 " <> tshow _ntfCount <> "\n# ntfCount1\n\
+      \simplex_smp_queues_notify_total2 " <> tshow notifierCount <> "\n# ntfCount2\n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:259]:
```suggestion
      \# HELP simplex_smp_queues_notify_total1 Total number of stored queues with notification flag (first type of count).\n\
      \# TYPE simplex_smp_queues_notify_total1 gauge\n\
      \simplex_smp_queues_notify_total1 " <> tshow _ntfCount <> "\n# ntfCount1\n\
      \\n\
      \# HELP simplex_smp_queues_notify_total2 Total number of stored queues with notification flag (second type of count).\n\
      \# TYPE simplex_smp_queues_notify_total2 gauge\n\
      \simplex_smp_queues_notify_total2 " <> tshow notifierCount <> "\n# ntfCount2\n\
```

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+    messages =
+      "# Messages\n\
+      \# --------\n\
+      \\n\
+      \# HELP simplex_smp_messages_sent Sent messages.\n\
+      \# TYPE simplex_smp_messages_sent counter\n\
+      \simplex_smp_messages_sent " <> tshow _msgSent <> "\n# msgSent\n\
+      \\n\
+      \# HELP simplex_smp_messages_errors Total number of messages errors by type.\n\
+      \# TYPE simplex_smp_messages_errors counter\n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:200]:
```suggestion
      \# HELP simplex_smp_messages_sent_errors Total number of messages errors by type.\n\
      \# TYPE simplex_smp_messages_sent_errors counter\n\
```

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+      \# TYPE simplex_smp_queues_weekly gauge\n\
+      \simplex_smp_queues_weekly " <> T.pack (weekCount ps) <> "\n# weekMsgQueues\n\
+      \\n\
+      \# HELP simplex_smp_queues_monthly Monthly active queues.\n\
+      \# TYPE simplex_smp_queues_monthly gauge\n\
+      \simplex_smp_queues_monthly " <> T.pack (monthCount ps) <> "\n# monthMsgQueues\n\
+      \\n\
+      \# HELP simplex_smp_queues_notify_daily Daily active queues with notifications.\n\
+      \# TYPE simplex_smp_queues_notify_daily gauge\n\
+      \simplex_smp_queues_notify_active_daily " <> T.pack (dayCount psNtf) <> "\n# dayCountNtf\n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:155]:
```suggestion
      \# HELP simplex_smp_queues_notify_daily Daily active queues with notifications.\n\
      \# TYPE simplex_smp_queues_notify_daily gauge\n\
      \simplex_smp_queues_notify_daily " <> T.pack (dayCount psNtf) <> "\n# dayCountNtf\n\
```

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+      \simplex_smp_relay_messages_errors{source=\"all\",type=\"connect\"} " <> tshow (_pErrorsConnect _pMsgFwds) <> "\n# pMsgFwds_pErrorsConnect\n\
+      \simplex_smp_relay_messages_errors{source=\"all\",type=\"compat\"} " <> tshow (_pErrorsCompat _pMsgFwds) <> "\n# pMsgFwds_pErrorsCompat\n\
+      \simplex_smp_relay_messages_errors{source=\"all\",type=\"other\"} " <> tshow (_pErrorsOther _pMsgFwds) <> "\n# pMsgFwds_pErrorsOther\n\
+      \simplex_smp_relay_messages_errors{source=\"own\",type=\"connect\"} " <> tshow (_pErrorsConnect _pMsgFwdsOwn) <> "\n# pMsgFwdsOwn_pErrorsConnect\n\
+      \simplex_smp_relay_messages_errors{source=\"own\",type=\"compat\"} " <> tshow (_pErrorsCompat _pMsgFwdsOwn) <> "\n# pMsgFwdsOwn_pErrorsCompat\n\
+      \simplex_smp_relay_messages_errors{source=\"own\",type=\"other\"} " <> tshow (_pErrorsOther _pMsgFwdsOwn) <> "\n# pMsgFwdsOwn_pErrorsOther\n\
+      \\n\
+      \# HELP simplex_smp_relay_messages Relay messages statistics.\n\
+      \# TYPE simplex_smp_relay_messages counter\n\
+      \simplex_smp_relay_messages_received " <> tshow _pMsgFwdsRecv <> "\n# pMsgFwdsRecv\n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:336]:
```suggestion
      \# HELP simplex_smp_relay_messages_received Relay messages statistics.\n\
      \# TYPE simplex_smp_relay_messages_received counter\n\
      \simplex_smp_relay_messages_received " <> tshow _pMsgFwdsRecv <> "\n# pMsgFwdsRecv\n\
```

File: src/Simplex/Messaging/Server/Prometheus.hs (hunk 6 above)
Comment (epoberezkin) [reply src/Simplex/Messaging/Server/Prometheus.hs:155]: corrections

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+      \# TYPE simplex_smp_subscribtion_successes_batch counter\n\
+      \simplex_smp_subscribtion_successes_batch " <> tshow _qSubAllB <> "\n# qSubAllB\n\
+      \\n\
+      \# HELP simplex_smp_subscribtion_end Ended subscriptions.\n\
+      \# TYPE simplex_smp_subscribtion_end counter\n\
+      \simplex_smp_subscribtion_end " <> tshow _qSubEnd <> "\n# qSubEnd\n\
+      \\n\
+      \# HELP simplex_smp_subscribtion_end_batch Batched ended subscriptions.\n\
+      \# TYPE simplex_smp_subscribtion_end_batch counter\n\
+      \simplex_smp_subscribtion_end " <> tshow _qSubEndB <> "\n# qSubEndB\n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:183]:
```suggestion
      \# HELP simplex_smp_subscribtion_end_batch Batched ended subscriptions.\n\
      \# TYPE simplex_smp_subscribtion_end_batch counter\n\
      \simplex_smp_subscribtion_end_batch " <> tshow _qSubEndB <> "\n# qSubEndB\n\
```

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+      \\n\
+      \# HELP simplex_smp_messages_ntf_errors Error events with notification messages (to ntf server). \n\
+      \# TYPE simplex_smp_messages_ntf_errors counter\n\
+      \simplex_smp_messages_ntf_errors{type=\"nosub\"} " <> tshow _msgNtfNoSub <> "\n# msgNtfNoSub\n\
+      \simplex_smp_messages_ntf_errors{type=\"lost\"} " <> tshow _msgNtfLost <> "\n# msgNtfLost\n\
+      \simplex_smp_messages_ntf_errors{type=\"expired\"} " <> tshow _msgNtfExpired <> "\n# msgNtfExpired\n\
+      \\n\
+      \# HELP simplex_smp_subscriptions_ntf_requests Subscription requests with notification flag (from ntf server). \n\
+      \# TYPE simplex_smp_subscriptions_ntf_requests counter\n\
+      \simplex_smp_subscriptions_ntf_requests " <> tshow _ntfSub <> "\n# ntfSub\n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:284]:
```suggestion
      \# HELP simplex_smp_subscription_ntf_requests Subscription requests with notification flag (from ntf server). \n\
      \# TYPE simplex_smp_subscription_ntf_requests counter\n\
      \simplex_smp_subscription_ntf_requests " <> tshow _ntfSub <> "\n# ntfSub\n\
```

File: src/Simplex/Messaging/Server/Prometheus.hs [hunk 10]
```diff
@@ -0,0 +1,378 @@
...
+      \\n\
+      \# HELP simplex_smp_subscriptions_ntf_requests_batch Batched subscription requests with notification flag (from ntf server). \n\
+      \# TYPE simplex_smp_subscriptions_ntf_requests_batch counter\n\
+      \simplex_smp_subscriptions_ntf_requests_batch " <> tshow _ntfSubB <> "\n# ntfSubB\n\
```
Comment (shumvgolove) [inline src/Simplex/Messaging/Server/Prometheus.hs:288]:
```suggestion
      \# HELP simplex_smp_subscription_ntf_requests_batch Batched subscription requests with notification flag (from ntf server). \n\
      \# TYPE simplex_smp_subscription_ntf_requests_batch counter\n\
      \simplex_smp_subscription_ntf_requests_batch " <> tshow _ntfSubB <> "\n# ntfSubB\n\
```

**Pull request merged**